src/tests/ftest/soak/soak.py

```diff
@@ -20,287 +20,433 @@
+    both the DAOS and MPI-IO drivers.
+
     """
 
     def job_done(self, args):
-        """
-        This is a callback function called when a job is done
+        """Call this function when a job is done.
 
-        handle --which job, i.e. the job ID
-        state  --string indicating job completion status
+        Args:
+            args (list):handle --which job, i.e. the job ID,
+                        state  --string indicating job completion status
         """
-
         self.soak_results[args["handle"]] = args["state"]
 
+    def create_pool(self, path):
+        """Create a pool that the various tests use for storage.
 
-    def create_pool(self):
-        """
-        Creates a pool that the various tests use for storage.
-        """
+        Args:
+            path: pool params from yaml file  /run/pool/* is default
+        Returns:
+            object: TestPool object
 
-        createmode = self.params.get("mode", '/run/pool1/createmode/*/')
-        createuid = os.geteuid()
-        creategid = os.getegid()
-        createsetid = self.params.get("setname", '/run/pool1/createset/')
-        createsize = self.params.get("size", '/run/pool1/createsize/')
-        self.createsvc = self.params.get("svcn", '/run/pool1/createsvc/')
+        """
+        # Create a pool
+        pool = TestPool(self.context, self.log)
+        pool.get_params(self, path)
+        pool.create()
+        self.log.info("Valid Pool UUID is %s", pool.uuid)
+
+        # Check that the pool was created
+        self.assertTrue(
+            pool.check_files(self.hostlist_servers),
+            "Pool data not detected on servers")
+        return pool
+
+    def destroy_pool(self, pool):
+        """Destroy the specified pool - TO DO."""
+        pass
+
+    def remote_copy(self, hostlist, remote_dir, local_dir):
+        """Copy files from remote dir to local dir.
+
+        This is a temporary method and will be replaced by
+        clush in general_utils
+        Args:
+                hostlist (list): list of remote nodes
+                remote_dir (str): remote directory of files
+                local_dir (str): local directory
+
+        Returns:
+            status: bool
 
-        self.pool = DaosPool(self.context)
-        self.pool.create(createmode, createuid, creategid,
-                         createsize, createsetid, None, None,
-                         self.createsvc)
+        """
+        this_host = socket.gethostname()
+        # Copy logfiles from non-empty client directories
+        command = "clush -w {} -B -S \"{}\"".format(
+            NodeSet.fromlist(hostlist),
+            "if [ ! -z \\\"\\$(ls -A {0})\\\" ]; then "
+            "scp -p -r {0}/ \\\"{1}:'{2}/'\\\" && rm -rf {0}/*; fi".format(
+                remote_dir, this_host, local_dir))
+        status = process.run(command, timeout=300)
+        return status
+
+    def create_ior_cmdline(self, job_params, job_spec, pool):
```

Comment:
Is this still needed? Now we have get_launch_command in ior_utils.py. Can we just use that directly?